pkg/cloud/services/networking/trunk.go

```diff
@@ -95,6 +95,9 @@ func (s *Service) DeleteTrunk(eventObject runtime.Object, portID string) error {
 				record.Eventf(eventObject, "SuccessfulDeleteTrunk", "Trunk %s with id %s did not exist", trunkInfo[0].Name, trunkInfo[0].ID)
 				return true, nil
 			}
+			if capoerrors.IsConflict(err) {
+				return false, err
```

Comment:
`util.PollImmediate` ultimately calls this function in k8s apimachinery with `immediate` as `true`:

https://github.com/kubernetes/apimachinery/blob/86c0c0f8c8e2ad6601321a412cddb1f99ff6fae5/pkg/util/wait/wait.go#L579-L587

Since the `err` returned by `runConditionWithCrashProtectionWithContext` - our function right here - is non-nil in the case of a conflict, we will still treat the conflict as an error. The first argument as `false` is never considered. So there is no retry.

So this change looks like a no-op? Should the `err` be `nil`?

Reply:
If `err` is set `nil`, then the user has to enable trunk in order for it to work. 
spec:
  template:
    spec:
      # Added for Testing purposes
      trunk: true
      ports:
      - description: "Port 1, trunk=true"
        trunk: true

Reply:
If trunk is enabled in the environment, but the port does not have a trunk configured, we still call `DeleteTrunk`, but I would expect that `len(trunkInfo) == 0` (where `trunkInfo` is the result of doing a list trunks against the OpenStack API) in the lines above:

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/c0819a496c023e905bb865e2c9907d642bb69070/pkg/cloud/services/networking/trunk.go#L88-L90

We should be returning `nil` and therefore never reach the code block that you have changed. Is this not happening somehow?

Reply:
You were right. The failure seems to be flaky case. 
One test passed, and, nothing is logged, implying that the code did not reach that part. 
removing the log now.